### PR TITLE
(CT-164) Fix useless error when puppet-access unconfigured 

### PR DESCRIPTION
**Pull request by BobosilaVictor**
Puppet-access provides a useless error when unconfigured

### CODE REVIEW

File: app/puppet-access/login.go [hunk 1]
```diff
@@ -6,13 +6,52 @@ import (
...
+		return "", fmt.Errorf("The response did not contain a token. Rerun with --debug to see full body")
+	}
+
+	return resp.Payload.Token, nil
+}
+
 // Login method requests a token from the server
-func (puppetAccess *PuppetAccess) Login() (string, error) {
+func (puppetAccess *PuppetAccess) Login() (*operations.LoginOK, error) {
 	client, err := puppetAccess.Client.GetClient()
```
Comment (ciprianbadescu) [inline app/puppet-access/login.go:52]: this should be moved to `LoginWithErrorDetails` and `Login` method could be removed

**Review by ciprianbadescu — APPROVED** [review]

File: app/puppet-access/login.go [hunk 2]
```diff
@@ -6,8 +6,31 @@ import (
...
+			return fmt.Errorf("Recieved an error while trying to request the token. The error was %v: %v", du.Payload.Kind, du.Payload.Msg)
+		}
+	}
+	if du, ok := err.(*operations.LoginDefault); ok {
+		log.Debug(err.Error())
+		return fmt.Errorf("Recieved an error while trying to request the token. The error with status code %d was %v", du.Code(), du.Payload)
+	}
+
+	log.Debug(err.Error())
+	return fmt.Errorf("Unkown error")
```
Comment (GabrielNagy) [inline app/puppet-access/login.go:31]: typo: Unknown

File: app/puppet-access/login.go [hunk 3]
```diff
@@ -6,8 +6,31 @@ import (
...
+	}
+	if du, ok := err.(*operations.LoginBadRequest); ok {
+		if du.Payload != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("Recieved an error while trying to request the token. The error was %v: %v", du.Payload.Kind, du.Payload.Msg)
+		}
+	}
+	if du, ok := err.(*operations.LoginDefault); ok {
+		log.Debug(err.Error())
+		return fmt.Errorf("Recieved an error while trying to request the token. The error with status code %d was %v", du.Code(), du.Payload)
```
Comment (GabrielNagy) [inline app/puppet-access/login.go:27]: typo: Received

File: app/puppet-access/login.go [hunk 4]
```diff
@@ -6,8 +6,31 @@ import (
...
+	if du, ok := err.(*operations.LoginUnauthorized); ok {
+		if du.Payload != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("Recieved an error while trying to request the token. The error was %v: %v", du.Payload.Kind, du.Payload.Msg)
+		}
+	}
+	if du, ok := err.(*operations.LoginBadRequest); ok {
+		if du.Payload != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("Recieved an error while trying to request the token. The error was %v: %v", du.Payload.Kind, du.Payload.Msg)
```
Comment (GabrielNagy) [inline app/puppet-access/login.go:22]: typo: Received

File: app/puppet-access/login.go [hunk 5]
```diff
@@ -6,8 +6,31 @@ import (
...
 	"github.com/puppetlabs/pe-sdk-go/app/puppet-access/api/client/operations"
 	"github.com/puppetlabs/pe-sdk-go/app/puppet-access/api/models"
+	"github.com/puppetlabs/pe-sdk-go/log"
 )
 
+func errorHandling(err error) error {
+	if du, ok := err.(*operations.LoginUnauthorized); ok {
+		if du.Payload != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("Recieved an error while trying to request the token. The error was %v: %v", du.Payload.Kind, du.Payload.Msg)
```
Comment (GabrielNagy) [inline app/puppet-access/login.go:16]: typo: Received

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**